File: sdk/samples/all in on code/Visual Studio 2010/VBAzureWorkflow4ServiceBus/Client/WorkflowServiceContract.vb
Imports System.ServiceModel

<ServiceContract()> _
Public Interface IProcessDataWorkflowService
	' Methods
	<OperationContract()> _
	Function ProcessData(ByVal request As ProcessDataRequest) As ProcessDataResponse
End Interface

Public Interface IProcessDataWorkflowServiceChannel
	Inherits IProcessDataWorkflowService, IClientChannel
End Interface

' The message contract.
<MessageContract(IsWrapped:=False)> _
Public Class ProcessDataRequest
	Public Sub New()
	End Sub

	Public Sub New(ByVal int As Integer?)
		Me.int = int
	End Sub

	<MessageBodyMember(Namespace:="http://schemas.microsoft.com/2003/10/Serialization/", Order:=0)> _
	Public int As Integer?
End Class

<MessageContract(IsWrapped:=False)> _
Public Class ProcessDataResponse
	Public Sub New()
	End Sub

	Public Sub New(ByVal [string] As String)
		Me.string = [string]
	End Sub

	<MessageBodyMember(Namespace:="http://schemas.microsoft.com/2003/10/Serialization/", Order:=0)> _
 Public [string] As String
End Class

File: sdk/samples/all in on code/Visual Studio 2010/CSAzureWorkflow4ServiceBus/Client/WorkflowServiceContract.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;

namespace Client
{
	[ServiceContract]
	public interface IProcessDataWorkflowService
	{
		[OperationContract]
		ProcessDataResponse ProcessData(ProcessDataRequest request);
	}

	public interface IProcessDataWorkflowServiceChannel :IProcessDataWorkflowService, IClientChannel
	{
	}

	// The message contract.
	[MessageContract(IsWrapped = false)]
	public partial class ProcessDataRequest
	{
		[MessageBodyMember(Namespace = "http://schemas.microsoft.com/2003/10/Serialization/", Order = 0)]
		public System.Nullable<int> @int;

		public ProcessDataRequest()
		{
		}

		public ProcessDataRequest(System.Nullable<int> @int)
		{
			this.@int = @int;
		}
	}

	[MessageContract(IsWrapped = false)]
	public partial class ProcessDataResponse
	{
		[MessageBodyMember(Namespace = "http://schemas.microsoft.com/2003/10/Serialization/", Order = 0)]
		public string @string;

		public ProcessDataResponse()
		{
		}

		public ProcessDataResponse(string @string)
		{
			this.@string = @string;
		}
	}
}
